hadoop-hdds/common/src/test/java/org/apache/hadoop/utils/db/TestRDBStore.java

```diff
@@ -289,4 +290,41 @@ public void testRocksDBCheckpointCleanup() throws Exception {
           checkpoint.getCheckpointLocation()));
     }
   }
+
+  /**
+   * Not strictly a unit test. Just a confirmation of the expected behavior
+   * of RocksDB keyMayExist API.
+   * @throws Exception if unable to read from RocksDB.
+   */
+  @Test
+  public void testRocksDBKeyMayExistApi() throws Exception {
+    try (RDBStore newStore =
+             new RDBStore(folder.newFolder(), options, configSet)) {
+      RocksDB db = newStore.getDb();
+
+      for (int i = 0; i < 100; i++) {
+        db.put(StringUtils.getBytesUtf16("key" + i),
+            StringUtils.getBytesUtf16("Value" + i));
+      }
+
+      long start = System.nanoTime();
+      for (int i = 0; i < 50; i++) {
+        Assert.assertTrue(db.get(
+            StringUtils.getBytesUtf16("key" + i))!= null);
+      }
+      long end = System.nanoTime();
+      long keyGetLatency = end - start;
+
+      start = System.nanoTime();
+      for (int i = 50; i < 100; i++) {
+        Assert.assertTrue(db.get(
+            StringUtils.getBytesUtf16("key" + i))!= null);
+      }
```

Comment:
I have not understood how this test is actually testing the API?

Reply:
Updated the UT. Hopefully this is clearer